tests/sentry/rules/processing/test_buffer_processing.py

```diff
@@ -0,0 +1,294 @@
+        )
+        self.create_event(self.project_two.id, FROZEN_TIME, "group-3", self.environment2.name)
+        self.create_event(self.project_two.id, FROZEN_TIME, "group-3", self.environment2.name)
+        self.create_event(self.project_two.id, FROZEN_TIME, "group-3", self.environment2.name)
+        assert self.event3.group
+        self.group3 = self.event3.group
+
+        self.rule4 = self.create_project_rule(
+            project=self.project_two, condition_data=[event_frequency_percent_condition]
+        )
+        self.event4 = self.create_event(self.project_two.id, FROZEN_TIME, "group-4")
+        assert self.event4.group
+        self.create_event(self.project_two.id, FROZEN_TIME, "group-4")
+        self._make_sessions(60, project=self.project_two)
+        self.group4 = self.event4.group
+
+        self.rulegroup_event_mapping_two = {
+            f"{self.rule3.id}:{self.group3.id}": {self.event3.event_id},
+            f"{self.rule4.id}:{self.group4.id}": {self.event4.event_id},
+        }
+        self.buffer_mapping = {
+            self.project.id: self.rulegroup_event_mapping_one,
+            self.project_two.id: self.rulegroup_event_mapping_two,
+        }
+        buffer.backend.push_to_sorted_set(key=PROJECT_ID_BUFFER_LIST_KEY, value=self.project.id)
+        buffer.backend.push_to_sorted_set(key=PROJECT_ID_BUFFER_LIST_KEY, value=self.project_two.id)
+
+    def _push_base_events(self) -> None:
+        self.push_to_hash(self.project.id, self.rule1.id, self.group1.id, self.event1.event_id)
+        self.push_to_hash(self.project.id, self.rule2.id, self.group2.id, self.event2.event_id)
+        self.push_to_hash(self.project_two.id, self.rule3.id, self.group3.id, self.event3.event_id)
+        self.push_to_hash(self.project_two.id, self.rule4.id, self.group4.id, self.event4.event_id)
+
+
+class ProcessBufferTest(ProcessDelayedAlertConditionsTestBase):
+    @patch("sentry.rules.processing.buffer_processing.process_in_batches")
+    def test_fetches_from_buffer_and_executes(self, mock_process_in_batches):
+        self._push_base_events()
+        # To get the correct mapping, we need to return the correct
+        # rulegroup_event mapping based on the project_id input
+        process_buffer()
+
+        for project, rule_group_event_mapping in (
+            (self.project, self.rulegroup_event_mapping_one),
+            (self.project_two, self.rulegroup_event_mapping_two),
+        ):
+            assert mock_process_in_batches.call_count == 2
+
+        project_ids = buffer.backend.get_sorted_set(
+            PROJECT_ID_BUFFER_LIST_KEY, 0, self.buffer_timestamp
+        )
+        assert project_ids == []
+
+    @patch("sentry.rules.processing.buffer_processing.process_in_batches")  # False option
+    @patch(
+        "sentry.rules.processing.delayed_processing.DelayedRule.option",
+        "delayed_processing.emit_logs",
+    )
+    def test_skips_processing_with_option(self, mock_process_in_batches):
```

Comment:
the only new test

Reply:
how is the option set to false here?

Reply:
the `delayed_processing.emit_logs` option is defaulted to False. perhaps it is a bit hacky 🤔

Reply:
ahh I see ok